src/data_provider/src/packages/packagesWindowsParserHelper.h

```diff
@@ -43,28 +44,39 @@ namespace PackageWindowsHelper
         {
             std::set<std::string> hotfixes;
             Utils::Registry root{key, subKey, KEY_WOW64_64KEY | KEY_ENUMERATE_SUB_KEYS | KEY_READ};
-            const auto packages{root.enumerate()};
-            for (const auto& package : packages)
+            const auto callback
             {
-                if (Utils::startsWith(package, "Package_"))
+                [&key, &subKey, &hotfixes](const std::string& package)
                 {
-                    std::string value;
-                    Utils::Registry packageReg{key, subKey + "\\" + package, KEY_WOW64_64KEY | KEY_READ};
-                    if (packageReg.string("InstallLocation", value))
+                    if (Utils::startsWith(package, "Package_"))
                     {
-                        const auto hfValue { extractHFValue(value) };
+                        auto hfValue { extractHFValue(package) };
```

Comment:
Based on https://github.com/wazuh/wazuh/blob/0e1e79a070cb7775baa441d376a0bad3d65c14be/src/wazuh_modules/syscollector/syscollector_windows.c#L1002 the block from line 53 to 57 should be in the else of the conditional on line 58 ?

Reply:
flow is:
1- try to extract KB value from package name.
2- if 1 failed and the package name contains "RollupFix", try to extract the KB value from "InstallLocation" value.